### PR TITLE
PyQt5 sinogram visualisation 

### PR DESCRIPTION
**Pull request by robbietuk**
Sets up the basic PyQt5 UI and uses MatPlotLib for image show. Currently, takes command line argument for projection data path.

Remaining todo:

v0.2
- [x] File browser built into UI
- [x] Simplify UI layout. 
1. - [ ] Use pages to switch between `sinogram` and `viewgram`](https://realpython.com/python-pyqt-layout/#creating-a-stack-of-widgets)
2. - [ ] [Improve stretch](https://realpython.com/python-pyqt-layout/#managing-space-in-box-layouts)
- [x] Simplify controll (slider and scrollbox) backend
- [ ] Improve MatPlotLib visualisation interface OR use alternative libraries (_ABANDONED_)

v0.3
- [ ] Add 3D scanner view with 2D slice visualisation (_ABANDONED_)

--------
Development version screen share.

image

### CODE REVIEW

**Comment by robbietuk** [comment]
New updated version.
image

This is probably the last mayor update I will do to this script. It is a useful tool IMO.

**Comment by KrisThielemans** [comment]
Thanks! How would one run this from an existing python session, ideally with projection data already loaded?

File: examples/python/projdata_visualisation/ProjDataVisualisation.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+
+def OpenProjDataVisualisation(filename=None, projdata=None):
+    app = QApplication([])
+    gallery = ProjDataVisualisationWidgetGallery()
+    
+    if isinstance(filename, str):
+        gallery.load_projdata(filename)
+    elif projdata is not None:
+        gallery.set_projdata(projdata)
```
Comment (KrisThielemans) [inline examples/python/projdata_visualisation/ProjDataVisualisation.py:250]: that's rather non-intuitive. I'd probably just remove the `filename` argument, or (probably better) do
```suggestion
def OpenProjDataVisualisation(projdata=None):
    app = QApplication([])
    gallery = ProjDataVisualisationWidgetGallery()
    
    if isinstance(projdata, str):
        gallery.load_projdata(projdata)
    elif  isinstance(projdata, str):
        gallery.set_projdata(projdata)
     elif projdata is not None:
       throw runtimeerrorsomehow
```

Comment (KrisThielemans) [reply examples/python/projdata_visualisation/ProjDataVisualisation.py:250]: could also use a docstring if you don't mind.

**Review by KrisThielemans — CHANGES_REQUESTED** [review]
great thanks. one small request... Finally, can you add a line in release_5.1.htm? Then I'll leave you alone :-)

**Comment by robbietuk** [comment]
One moment before merge, fixing a bug.

**Comment by robbietuk** [comment]
Done now. I fixed a bug if the filename was invalid. Rather than crash, it now reports it to the GUI.

**Review by KrisThielemans — APPROVED** [review]
thanks!
(no release notes, but I'll do that later)

**Comment by robbietuk** [comment]
> thanks! (no release notes, but I'll do that later)

I see you have done that. Thank you. Hopefully it can be useful!

**Pull request merged**